scalableminds/webknossos
# Analytics in Postgres

- Adds tracking of analytics events in our own postgres database
- Adds the `POST /api/analytics/ingest` route to accept events from other wk instances, just like the current event-relay does

### Steps to test:
- `fetch("/api/analytics/ingest", {method:"POST", headers:{"content-type": "application/json"}, body: JSON.stringify({"events":[{"event_type":"open_annotation","user_id":"123456789012345678901234","time":"1706510609682","user_properties":{"organization_id":"123456789012345678901234","is_organization_admin":false,"is_superuser":false,"webknossos_uri":"https://webknossos.org"},"event_properties":{"annotation_id":"123456789012345678901234","annotation_owner_multiuser_id":"123456789012345678901234","annotation_dataset_id":"123456789012345678901234"},"session_id":1706510609681}], "api_key": "dingdong"})})` (also try with other `webknossos_uri` and `api_key`)
- Click around in wk and check the postgres database

### TODOs:
- [x] Add migration

------
(Please delete unneeded items, merge only when none are left open)
